app/main.py

```diff
@@ -1,3 +1,34 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+from customer import Customer
+from shop import Shop
+
+
+def shop_trip() -> None:
```

Comment:
The `shop_trip` function should include functionality for printing a purchase receipt with the current time and ensuring the customer returns home with the remaining money, as specified in the task description .